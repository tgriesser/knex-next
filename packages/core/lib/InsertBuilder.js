"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Grammar_1 = require("./Grammar");
const datatypes_1 = require("./data/datatypes");
class InsertBuilder {
    constructor(ast = datatypes_1.insertAst) {
        this.ast = ast;
        this.dialect = null;
        this.grammar = new Grammar_1.Grammar();
    }
    values(toInsert) { }
    select(subQuery) { }
    inBatchesOf(value) { }
    insertInto(tableName) {
        return this.chain(ast => ast.set("table", tableName));
    }
    insertGetId() {
        return this.chain(ast => {
            return ast;
        });
    }
    getAst() {
        return this.ast;
    }
    toOperation() {
        return this.grammar.toOperation(this.ast);
    }
    chain(fn) {
        this.ast = fn(this.ast);
        return this;
    }
    log(msg) {
        console.log(msg);
    }
    error(err) {
        console.error(err);
    }
    warn(warning) {
        console.warn(warning);
    }
}
exports.InsertBuilder = InsertBuilder;
