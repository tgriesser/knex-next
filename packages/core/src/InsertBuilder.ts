import { ChainFnInsert, SubQueryArg, TRawNode, Maybe, ExecutableBuilder } from "./data/types";
import { Grammar } from "./Grammar";
import { insertAst } from "./data/structs";
import { IBuilder } from "./contracts/Buildable";
import { SelectBuilder } from "./SelectBuilder";
import { Connection } from "./Connection";

export interface InsertBuilder extends ExecutableBuilder {}

export class InsertBuilder<T = { [columnName: string]: any }> implements IBuilder {
  public readonly dialect = null;

  protected grammar = new Grammar();

  constructor(protected ast = insertAst) {}

  protected connection: Maybe<Connection> = null;

  clearValues() {
    return this.chain(ast => ast.set("values", insertAst.values));
  }

  into(tableName: string) {
    return this.chain(ast => ast.set("table", tableName));
  }

  columns(...columnName: string[]) {
    return this.chain(ast => ast.set("columns", ast.columns.concat(columnName)));
  }

  values(toInsert: T | T[]) {
    return this.chain(ast => ast.set("values", ast.values.concat(toInsert)));
  }

  select(arg: SubQueryArg | TRawNode) {
    if (typeof arg === "function") {
      return this.chain(ast => ast.set("select", this.subQuery(arg)));
    }
    return this;
  }

  inBatchesOf(value: number) {
    return this.chain(ast => ast.set("chunkSize", value));
  }

  getAst() {
    return this.ast;
  }

  protected subQuery(arg: SubQueryArg) {
    const builder = new SelectBuilder();
    arg.call(builder, builder);
    return builder.getAst();
  }

  protected chain(fn: ChainFnInsert) {
    this.ast = fn(this.ast);
    return this;
  }
}
