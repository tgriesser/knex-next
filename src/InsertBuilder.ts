import { ChainFnInsert, SubQueryArg } from "./data/types";
import { Grammar } from "./Grammar";
import { insertAst } from "./data/datatypes";
import { Loggable } from "./contracts/Loggable";

export class InsertBuilder<T = { [columnName: string]: any }>
  implements Loggable {
  public readonly dialect = null;

  protected grammar = new Grammar();

  constructor(protected ast = insertAst) {}

  values(toInsert: T | T[]) {}

  select(subQuery: SubQueryArg) {}

  inBatchesOf(value: number) {}

  insertInto(tableName: string) {
    return this.chain(ast => ast.set("table", tableName));
  }

  insertGetId() {
    return this.chain(ast => {
      return ast;
    });
  }

  getAst() {
    return this.ast;
  }

  toOperation() {
    return this.grammar.toOperation(this.ast);
  }

  protected chain(fn: ChainFnInsert) {
    this.ast = fn(this.ast);
    return this;
  }

  log(msg: string) {
    console.log(msg);
  }

  error(err: Error) {
    console.error(err);
  }

  warn(warning: string | Error) {
    console.warn(warning);
  }
}
