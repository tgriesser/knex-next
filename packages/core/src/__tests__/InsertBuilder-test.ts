import { InsertBuilder } from "../InsertBuilder";
import { commonInsertTests } from "../testing/common-insert-tests";

describe("@knex/core - InsertBuilder", () => {
  commonInsertTests(() => new InsertBuilder());
});
