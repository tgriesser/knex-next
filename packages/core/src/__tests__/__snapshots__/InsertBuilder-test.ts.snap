// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@knex/core - InsertBuilder insert: select 1`] = `
Immutable.InsertOperation {
  "__operation": "INSERT",
  "table": "someTable",
  "chunkSize": null,
  "columns": Immutable.List [
    "a",
  ],
  "values": Immutable.List [],
  "select": Immutable.SelectNodes {
    "__operation": "SELECT",
    "from": "otherTable",
    "where": Immutable.List [],
    "select": Immutable.List [
      "a",
    ],
    "join": Immutable.List [],
    "having": Immutable.List [],
    "group": Immutable.List [],
    "order": Immutable.List [],
    "union": Immutable.List [],
    "limit": null,
    "offset": null,
    "alias": null,
    "distinct": false,
    "lock": null,
    "meta": Immutable.Map {},
  },
  "meta": Immutable.Map {},
}
`;

exports[`@knex/core - InsertBuilder insert: select 2`] = `
Object {
  "fragments": Array [
    "INSERT INTO someTable SELECT a FROM otherTable",
  ],
  "query": "INSERT INTO someTable SELECT a FROM otherTable",
  "sql": "INSERT INTO someTable SELECT a FROM otherTable",
  "values": Array [],
}
`;

exports[`@knex/core - InsertBuilder insert: values 1`] = `
Immutable.InsertOperation {
  "__operation": "INSERT",
  "table": "some_table",
  "chunkSize": null,
  "columns": Immutable.List [],
  "values": Immutable.List [],
  "select": null,
  "meta": Immutable.Map {},
}
`;

exports[`@knex/core - InsertBuilder insert: values 2`] = `
Object {
  "fragments": Array [
    "INSERT INTO some_table",
  ],
  "query": "INSERT INTO some_table",
  "sql": "INSERT INTO some_table",
  "values": Array [],
}
`;

exports[`@knex/core - InsertBuilder multiple inserts 1`] = `
Immutable.InsertOperation {
  "__operation": "INSERT",
  "table": "users",
  "chunkSize": null,
  "columns": Immutable.List [],
  "values": Immutable.List [
    Object {
      "email": "foo",
      "name": "tim",
    },
    Object {
      "email": "bar",
      "name": "kb",
    },
  ],
  "select": null,
  "meta": Immutable.Map {},
}
`;

exports[`@knex/core - InsertBuilder multiple inserts 2`] = `
Object {
  "fragments": Array [
    "INSERT INTO users",
  ],
  "query": "INSERT INTO users",
  "sql": "INSERT INTO users",
  "values": Array [],
}
`;
